File: backend/app/schemas/agent.py
from typing import Any

from pydantic import BaseModel, Field

from app.schemas.profile import ProfileData
from app.schemas.rag import SourceNode


class ModalitySignal(BaseModel):
    modality: str = ""
    signal: str = ""
    score: float = 0.0


class AgentIntentInfo(BaseModel):
    code: str = "general_consultation"
    label: str = "一般咨询"
    confidence: float = 0.0
    reason: str = ""


class FraudTypeInfo(BaseModel):
    code: str = "suspicious_script"
    label: str = "可疑诈骗话术"
    confidence: float = 0.0
    rationale: str = ""


class EvidenceItem(BaseModel):
    title: str = ""
    source: str = ""
    snippet: str = ""
    score: float = 0.0
    modality: str = ""
    kind: str = ""


class InterventionAction(BaseModel):
    label: str = ""
    description: str = ""
    priority: str = "medium"


class InterventionPlanInfo(BaseModel):
    level: str = "low"
    headline: str = ""
    summary: str = ""
    recommended_channel: str = ""
    actions: list[InterventionAction] = Field(default_factory=list)


class GuardianActionInfo(BaseModel):
    priority: str = "none"
    notice: str = ""
    triggered: bool = False
    target_role: str = ""
    message_template: str = ""
    next_step: str = ""
    checklist: list[str] = Field(default_factory=list)


class AgentReportInfo(BaseModel):
    title: str = "安全监测报告"
    executive_summary: str = ""
    findings: list[str] = Field(default_factory=list)
    disposition: str = ""
    recommended_actions: list[str] = Field(default_factory=list)


class AgentAnalyzeResponse(BaseModel):
    risk_level: str = "low"
    risk_score: float = 0.0
    intent: AgentIntentInfo = Field(default_factory=AgentIntentInfo)
    fraud_type: FraudTypeInfo = Field(default_factory=FraudTypeInfo)
    fraud_types: list[str] = Field(default_factory=list)
    evidence: list[EvidenceItem] = Field(default_factory=list)
    intervention_plan: InterventionPlanInfo = Field(default_factory=InterventionPlanInfo)
    modalities_received: list[str] = Field(default_factory=list)
    signals: list[ModalitySignal] = Field(default_factory=list)
    text_result: dict[str, Any] = Field(default_factory=dict)
    image_result: dict[str, Any] = Field(default_factory=dict)
    audio_result: dict[str, Any] = Field(default_factory=dict)
    rag_sources: list[SourceNode] = Field(default_factory=list)
    profile_summary: ProfileData = Field(default_factory=ProfileData)
    recommendations: list[str] = Field(default_factory=list)
    guardian_action_needed: bool = False
    guardian_action: GuardianActionInfo = Field(default_factory=GuardianActionInfo)
    report: AgentReportInfo = Field(default_factory=AgentReportInfo)
    summary: str = ""

File: frontend/src/types/agent.ts
import type { ProfileData } from '@/api/profile'
import type { RagSource } from '@/types/rag'

export interface AgentSignal {
  modality: string
  signal: string
  score: number
}

export interface GuardianActionInfo {
  priority: string
  notice: string
  triggered: boolean
  target_role: string
  message_template: string
  next_step: string
  checklist: string[]
}

export interface GuardianNotificationInfo {
  attempted: boolean
  sent: boolean
  status: string
  message: string
  recipient_masked: string
}

export interface AgentIntentInfo {
  code: string
  label: string
  confidence: number
  reason: string
}

export interface AgentFraudTypeInfo {
  code: string
  label: string
  confidence: number
  rationale: string
}

export interface AgentEvidenceItem {
  title: string
  source: string
  snippet: string
  score: number
  modality: string
  kind: string
}

export interface AgentInterventionAction {
  label: string
  description: string
  priority: string
}

export interface AgentInterventionPlan {
  level: string
  headline: string
  summary: string
  recommended_channel: string
  actions: AgentInterventionAction[]
}

export interface AgentReportInfo {
  title: string
  executive_summary: string
  findings: string[]
  disposition: string
  recommended_actions: string[]
}

export interface AgentAnalyzeResponse {
  risk_level: string
  risk_score: number
  intent: AgentIntentInfo
  fraud_type: AgentFraudTypeInfo
  fraud_types: string[]
  evidence: AgentEvidenceItem[]
  intervention_plan: AgentInterventionPlan
  modalities_received: string[]
  signals: AgentSignal[]
  text_result: Record<string, any>
  image_result: Record<string, any>
  audio_result: Record<string, any>
  rag_sources: RagSource[]
  profile_summary: ProfileData
  recommendations: string[]
  guardian_action_needed: boolean
  guardian_action: GuardianActionInfo
  guardian_notification: GuardianNotificationInfo
  user_ack_required: boolean
  report: AgentReportInfo
  summary: string
}
